bin/wasm-node/javascript/src/index-browser.ts

```diff
@@ -121,23 +125,200 @@ function trustedBase64Decode(base64: string): Uint8Array {
+            // The SCTP port (RFC8841)
+            // Note it's different from the "m=" line port value, which
+            // indicates the port of the underlying transport-layer protocol
+            // (UDP or TCP)
+            "a=sctp-port:5000" + "\n" +
+            // The maximum SCTP user message size (in bytes) (RFC8841)
+            "a=max-message-size:100000" + "\n" +
+            // A transport address for a candidate that can be used for connectivity checks (RFC8839).
+            "a=candidate:1 1 UDP 1 " + targetIp + " " + targetPort + " typ host" + "\n";
+
+        await pc.setRemoteDescription({ type: "answer", sdp: remoteSdp });
 
-    openOutSubstream: () => { throw new Error('Wrong connection type') }
-  };
+        console.log("REMOTE ANSWER: " + pc.remoteDescription!.sdp);
+    };
+
+    const dataChannels = new Map<number, RTCDataChannel>();
+    const addChannel = (dataChannel: RTCDataChannel, direction: 'inbound' | 'outbound') => {
+      const dataChannelId = dataChannel.id!;
+
+      dataChannel.onopen = () => {
+          console.log(`'${dataChannel.label}' opened, direction is ${direction}`);
+          config.onStreamOpened(dataChannelId, direction);
+      };
+
+      dataChannel.onerror = (error) => {
+          console.log(`'${dataChannel.label}' errored: ${error}`);
+          config.onStreamClose(dataChannelId);
+      };
+
+      dataChannel.onclose = () => {
+          console.log(`'${dataChannel.label}' closed`);
+          config.onStreamClose(dataChannelId);
+      };
+
+      dataChannel.onmessage = (m) => {
+          console.log(`new message on '${dataChannel.label}': '${m.data}'`);
+          // The `data` field is an `ArrayBuffer`.
+          config.onMessage(new Uint8Array(m.data), dataChannelId);
+      }
+
+      dataChannels.set(dataChannelId, dataChannel);
+    }
+
+    pc.ondatachannel = ({ channel }) => {
+      addChannel(channel, 'inbound')
+    };
+
+    // TODO: this system is a complete hack
+    let outId = 1;
+
+    // TODO: smoldot panics if we call onOpen before returning
+    setTimeout(() => {
+      config.onOpen({ type: 'multi-stream' });
+    }, 0);
+
+    return {
+      close: (streamId: number): void => {
+        if (streamId == null) {
+          pc.onconnectionstatechange = null;
+          pc.onnegotiationneeded = null;
+          pc.close();
+        } else {
+          const channel = dataChannels.get(streamId)!;
+          channel.onopen = null;
+          channel.onerror = null;
+          channel.onclose = null;
+          channel.onmessage = null;
+          channel.close();
+          dataChannels.delete(streamId);
+        }
+      },
+
+      send: (data: Uint8Array, streamId: number): void => {
+        const channel = dataChannels.get(streamId)!;
+        console.log("sent", data);
+        channel.send(data);
+      },
+
+      openOutSubstream: () => {
+        addChannel(pc.createDataChannel("data", { id: outId, negotiated: true }), 'outbound')
```

Comment:
only the first ("noise") data channel is negotiated and has predefined ID=1. other channels don't need neither of these things.